woocommerce/woocommerce
# Product Collection: Add loading indicator for client-side pagination

### Changes proposed in this Pull Request:





This PR introduces an animation(loading indicator) for client-side pagination in the Product Collection block. Inspired by the animation used in the Query Loop block, this feature aims to enhance user experience by displaying an animation if the navigation to a new page exceeds 400ms. For more information, refer to issue https://github.com/woocommerce/woocommerce/issues/44393.

Closes #44393



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a Product Collection block to a new post & choose any collection. 
2. Adjust the items per page in the toolbar controls to "3" to ensure only three products are displayed per page, creating multiple pages for navigation.
3. Publish the post & verify on Frontend that:
    1. An animation appears when changing pages through pagination if the new page's load time surpasses approximately 400ms, signaling that the next page is loading.
    2. Tip: Consider throttling the network speed to "Slow 3G" to simulate slower loading times.

**Sidenote:** It's recommended to test the frontend in incognito mode. This is because certain plugins might add a top navbar, potentially obscuring the animation. Testing in incognito mode helps ensure that the animation remains visible.

Here is a quick demo:

https://github.com/woocommerce/woocommerce/assets/16707866/c18d1700-e6c3-422e-bb1f-cb468382bead





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Product Collection: Add loading indicator for client-side pagination

#### Comment 

</details>
